std/stdio.d

```diff
@@ -1522,6 +1524,143 @@ for every line.
     }
 
 /**
+Creates a file with a randomly generated name and returns an open File to it.
+
+The name will start with the given prefix and be located in the given directory.
+If no directory is given, then it will default to the result of
+$(XREF file, tempDir).
+
+There is nothing special about the file created other than the fact that its
+name is randomly generated, and it defaults to being in the temp directory on
+the system. It will only be deleted if explicitly deleted or if it's in the temp
+directory, and the temp is cleared out (as happens on system startup or shutdown
+on some systems).
+
+Params:
+    prefix = Optional prefix for file name.
+    suffix = Optional suffix for file name (which also provides a way to give
+             the file an extension).
+    dir    = Directory of temporary file.
+  */
+    static File tempFile(string prefix = "", string suffix = "", string dir = "")
+    {
+        import std.exception : ErrnoException;
+        import std.file : isSymlink, tempDir;
+        import std.path : absolutePath, baseName, buildPath, dirName;
+
+        // For some reason, import std.file at the module level of std.stdio or
+        // even just inside of File results in a static assertion being hit in
+        // std.range.primitives.put. So, for now at least, we'll just assign
+        // dir here rather than using tempDir as a default argument so that the
+        // import can be local.
+        if(dir.empty)
+            dir = std.file.tempDir();
+
+        static string genTempName(string prefix, string suffix, string dir)
+        {
+            import std.ascii : digits, letters;
+            import std.random : rndGen;
+            rndGen.popFront();
+            auto name = new char[](prefix.length + 10 + rndGen.front % 5 + suffix.length);
+            name[0 .. prefix.length] = prefix;
+
+            char[letters.length + digits.length] chars;
+            chars[0 .. letters.length] = letters[];
+            chars[letters.length .. $] = digits[];
+
+            foreach(ref c; name[prefix.length .. $])
+            {
+                c = chars[rndGen.front % chars.length];
+                rndGen.popFront();
+            }
+
+            name[$ - suffix.length .. $] = suffix;
+            return buildPath(dir, name);
+        }
+
+        // Once should be enough given the randomness, but we'll give it three
+        // times to be safe.
+        foreach(i; 0 .. 3)
+        {
+            auto filename = genTempName(prefix, suffix, dir);
+
+            // O_EXCL does not support symbolic links.
+            if(dirName(filename).isSymlink)
+                filename = buildPath(absolutePath(dirName(filename), baseName(filename)));
+
+            import std.internal.cstring : tempCString;
+            version(Posix)
+            {
+                import core.sys.posix.sys.stat, core.sys.posix.fcntl;
+                enum flags = S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH;
```

Comment:
So, I take it that you're suggesting that this just be

    enum flags = S_IRUSR | S_IWUSR;

I guess that that makes sense. It's been so long since I wrote that that I don't even remember why I picked those permissions - probably because it was the least restrictive.